### DISCUSSION TITLE
Some questions I have for this project

### DISCUSSION THREAD

**Discussion by Raj2032** [gxi-rs/gxi#22] [General]
Hi, I have some questions regarding this project:
1. I understand it doesn't currently support Android, however since Android has webview support, am I able to make a web application, package it up using `cargo-apk` and then Android can display the app as a webview?
2. If compiled for Windows and Linux (when it supports Windows and Linux), would I get native performance or would there be performance overhead ~ would it be compiled as a web application running (like Electron)?
3. Is it retained or immediate or both?

**Comment by SamuraiCrow** [comment]
I can answer number 2.  The library that will be supported by GXI is GTK which is used by the LXDE, Gnome and XFCE desktop environments.  It will be as fast as one of those.  The Windows version will be a thin GTK compatibility layer respectively.

**Reply by Raj2032** [reply]
Thanks mate =D

**Comment by aniketfuryrocks** [comment]
I am designing the project in the most generic way possible, but due to some limitations in rust proc-macros, a few implementations have to be hard coded for each platform in the gxi library. In order to add support any widget library inexistence, all gxi needs is their rust bindings exporting structs or traits representing their native widget system.

You can think of gxi as a data structure, which helps you to manage a huge heap of native widgets. The gxi macro is the tool for devs to write that data structure with ease.

So which platforms are supported ?
Ans. All platforms where rust can compile to.

Catch ?
Ans. You have to get the platforms native gui bindings into the project.

Earlier I had both gtk and wasm bindings in the project, but after the last iteration I removed gtk, for the sake of simplicity.

Once I am comfortable with the stability of gxi macro, ill start to add all the bindings the community wants to have.

**Reply by Raj2032** [reply]
Ok thanks mate =D
